### ISSUE TITLE
Protection Groups - One Group to Manage Them All

### ISSUE THREAD

**Issue by Foxtrek64** [QuiltServerTools/HeyThatsMine#45]
**TL;DR with Scenario**: I have a city where I have a lot of protections on doors, chests, etc.. If someone joins or leaves this city, I must manually update every single protected door and chest. This is further complicated when I subdivide this into groups of people. The ability to create groups of users and to have protections use that group means only one thing needs to be updated to manage all of those protections. This suggestion is to add this Protection Group feature.

**Long Version**:
As it stands with HTM and other protection mods, if I have doors that will only be accessible by group 1, then I must get a list of everyone in that group and then manually go to each door/chest and add each person, one by one. Should that person leave group 1, then I must go to every protected block and manually update that protection to remove them. This can be a very tedious process when there are a large number of protected blocks.

So what I would like to propose is to add the ability to make groups a native feature of HTM. I can define groups, e.g. "royals", "guards", and "civilians", then grant appropriate permissions to those groups on those protections. Then if someone leaves the town guard, I can remove them from the group. When they interact with that protected block, it tests all explicit permissions (individuals added to the protection) followed by a check to see whether they are a member of a group that has been granted permission, and then allows the action or cancels it appropriately.

This also means that if I were to hire a replacement guard, I have to execute a single command to add them to the guards group and now they have access to all of the guard doors and guard chests. I don't have to run around and update every protection manually.

**Bonus features (stretch goals)**:
- Allow adding LP groups to a protection. For instance, I may want to set up a chest where members of the Donor group are allowed to use the chest, and that donors group is managed via LP.
- Per-block blacklist. If someone is a member of a group that is added to a protected block, but you don't want them to interact with that specific block, you can add them to the blacklist. The blacklist would be checked first and should support both users and Protection Groups.

**Other Thoughts**:
This could all be managed by LP, but I feel a custom group implementation may be more prudent for two reasons:
1. In servers where player land claims is a big feature (e.g. faction servers and roleplay servers), users would have to request admins to add or remove a person from a group, which creates a lot of unnecessary work for server staff.
2. In public servers, giving region owners the ability to manage LuckPerms groups is a bad idea. LP isn't set up to have "manage my own groups only" because it's assumed that only admins are going to be managing groups.
